### ISSUE TITLE
Allow stopping propagation implicitly from event/submit binding

### ISSUE THREAD

**Issue by cervengoc** [knockout/knockout#2209]
For example in `jQuery` we can prevent default behavior and stop propagation of an event by explicitly returning `false` from the event handler function. Could it be possible for `knockout` to implement the same behavior?

Currently we have two ways to control propagation, neither of them is very elegant.
- Explicitly introduce the `event` object in our handler function in the viewmodel, and call `stopPropagation` for example. I don't like this because it's rather a view-related thing which wouldn't belong to the viewmodel.
- Use the `*Bubble` bindings. This is a static way to control the propagation, so the handler doesn't have the possibility to decide whether the event should be propagated or not.

**Comment by mbest** [comment]
The return value is already used. If you return `true` from an event handler, it will cause the default behavior to run.

**Comment by cervengoc** [comment]
Yes, but I was primarily talking about propagation. We cannot dynamically control it now. I suggest the following overall behavior.

- If return value if `false` stop propagation and prevent default.
- If return value is `true` allow default and propagate.
- If return value is `undefined` prevent default behavior but do propagate the event.

Besides these, if an explicit `*Bubble` binding is defined, it should always override the above behavior accordingly.

Hope this makes sense, let me know if you need any more clarification.

**Comment by mbest** [comment]
That could work. @brianmhunt, what do you think?

**Comment by brianmhunt** [comment]
This is pretty intuitive because of jQuery so I think it works.

**Comment by cervengoc** [comment]
I'll have some time during the weekend to address this in a PR with some specs.

**Comment by mbest** [comment]
@cervengoc If you have the code for this, can you submit your pull request?

**Comment by cervengoc** [comment]
@mbest Unfortunately I have very little time nowadays. I'll hopefully be able to create a PR within next week or so.
